lib/Doctrine/DBAL/Driver/IBMDB2/DB2Connection.php

```diff
@@ -126,23 +121,35 @@ public function exec(string $statement) : int
     /**
      * {@inheritdoc}
      */
-    public function lastInsertId($name = null)
+    public function lastInsertId(?string $name = null) : string
     {
-        return db2_last_insert_id($this->conn);
+        if ($name !== null) {
+            // db2_last_insert_id() ignores the name parameter, and returns the last insert ID even if a sequence name
```

Comment:
> db2_last_insert_id() ignores the name parameter

I don't think it's correct. Besides the poorly designed PDO API, no other drivers support the sequence name in a call to the "last insert id" method. It should be implemented in a different way like `SELECT $name.currval FROM dual` (not part of this ticket).